src/os/bluestore/bluefs_types.cc

```diff
@@ -85,7 +85,7 @@ void bluefs_super_t::encode(bufferlist& bl) const
 {
   __u8 _version = 3;
   __u8 _compat = 1;
-  if (wal_v2) {
+  if (wal_v2 > 1) {
```

Comment:
write_super() sets wal_v2 to 1 for v2 support. Hence it should be either wal_v2 >= 1 or write_super has to be fixed.